Include/internal/pycore_long.h

```diff
@@ -110,25 +108,155 @@ PyAPI_FUNC(char*) _PyLong_FormatBytesWriter(
     int base,
     int alternate);
 
-/* Return 1 if the argument is positive single digit int */
+/* Long value tag bits:
+ * 0-1: Sign bits value = (1-sign), ie. negative=2, positive=0, zero=1.
+ * 2: Reserved for immortality bit
```

Comment:
I don't think we need an immortality bit here, we just need a static flag (immortality should be marked by the refcount and this marks if the object is static or not. Using this, we can do the static check at dealloc time to prevent the deallocation of the objects (for small ints and bools).

Plus, if we'd like to save that extra bit for a future use case, we can probably do something like:
```
#define SIGN_MASK 3
#define SIGN_ZERO 1
#define SIGN_NEGATIVE 2
#define SIGN_STATIC 3
#define NON_SIZE_BITS 3

static inline bool
_PyLong_IsStatic(const PyLongObject *op)
{
    return (op->long_value.lv_tag & SIGN_MASK) == SIGN_STATIC;
}
```

Thoughts? Also, cc @ericsnowcurrently